pkg/webhook/policy.go

```diff
@@ -182,23 +181,35 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 	}
 
 	res := resp.Results()
-	msgs := h.getDenyMessages(res, req)
-	if len(msgs) > 0 {
-		vResp := admission.ValidationResponse(false, strings.Join(msgs, "\n"))
+	denyMsgs, warnMsgs := h.getValidationMessages(res, req)
+
+	if len(denyMsgs) > 0 {
+		vResp := admission.ValidationResponse(false, strings.Join(denyMsgs, "\n"))
 		if vResp.Result == nil {
 			vResp.Result = &metav1.Status{}
 		}
+		if len(warnMsgs) > 0 {
+			vResp.Warnings = warnMsgs
+		}
 		vResp.Result.Code = http.StatusForbidden
 		requestResponse = denyResponse
 		return vResp
 	}
 
 	requestResponse = allowResponse
-	return admission.ValidationResponse(true, "")
+	vResp := admission.ValidationResponse(true, "")
+	if vResp.Result == nil {
+		vResp.Result = &metav1.Status{}
+	}
+	if len(warnMsgs) > 0 {
+		vResp.Warnings = warnMsgs
+		vResp.Result.Code = 299
```

Comment:
What is this magic code and where is it documented?

Reply:
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response

> Starting in v1.19, admission webhooks can optionally return warning messages that are returned to the requesting client in HTTP Warning headers with a warning code of 299.

Reply:
Awesome. Maybe we can create an `HttpStatusWarning` constant somewhere and link to the docs?

Reply:
added to this file. lmk if you think we should add somewhere else (in util maybe?)